aiobotocore/httpxsession.py

```diff
@@ -0,0 +1,305 @@
+from __future__ import annotations
+
+import asyncio
+import io
+import os
+import socket
+from typing import TYPE_CHECKING, Any, cast
+
+import botocore
+from botocore.awsrequest import AWSPreparedRequest
+from botocore.httpsession import (
+    MAX_POOL_CONNECTIONS,
+    EndpointConnectionError,
+    HTTPClientError,
+    ReadTimeoutError,
+    create_urllib3_context,
+    ensure_boolean,
+    get_cert_path,
+    logger,
+)
+from multidict import CIMultiDict
+
+import aiobotocore.awsrequest
+from aiobotocore._endpoint_helpers import _text
+
+# TODO: resolve future annotations thing
+
+
+try:
+    import httpx
+except ImportError:
+    httpx = None
+
+if TYPE_CHECKING:
+    from ssl import SSLContext
+
+
+class HttpxSession:
+    def __init__(
+        self,
+        verify: bool = True,
+        proxies: dict[str, str] | None = None,  # {scheme: url}
+        timeout: float | list[float] | tuple[float, float] | None = None,
+        max_pool_connections: int = MAX_POOL_CONNECTIONS,
+        socket_options: list[Any] | None = None,
+        client_cert: str | tuple[str, str] | None = None,
+        proxies_config: dict[str, str] | None = None,
+        connector_args: dict[str, Any] | None = None,
+    ):
+        if httpx is None:  # pragma: no cover
+            raise RuntimeError(
+                "Using HttpxSession requires httpx to be installed"
+            )
+        if proxies or proxies_config:
+            raise NotImplementedError(
+                "Proxy support not implemented with httpx as backend."
+            )
+
+        # TODO: handle socket_options
+        self._session: httpx.AsyncClient | None = None
+        conn_timeout: float | None
+        read_timeout: float | None
+
+        if isinstance(timeout, (list, tuple)):
+            conn_timeout, read_timeout = timeout
+        else:
+            conn_timeout = read_timeout = timeout
+        # must specify a default or set all four parameters explicitly
+        # 5 is httpx default default
+        self._timeout = httpx.Timeout(
+            5, connect=conn_timeout, read=read_timeout
```

Comment:
hmm, not sure about this.  while 5 is the default, it is be no means what people will be using in production.  Perhaps use aiobotocore.config.DEFAULT_TIMEOUT, or add some more params to AioConfig.  I like the idea of being able to specify the write and pool timeouts separately.

Reply:
added config options for them